src/functions.php

```diff
@@ -12,22 +12,29 @@
  * Creates a `Promise` which resolves with the stream data buffer
  *
  * @param ReadableStreamInterface $stream
+ * @param int|null $maxLength Maximum number of bytes to buffer or null for unlimited.
  * @return Promise\CancellablePromiseInterface Promise<string, Exception>
  */
-function buffer(ReadableStreamInterface $stream)
+function buffer(ReadableStreamInterface $stream, $maxLength = null)
 {
     // stream already ended => resolve with empty buffer
     if (!$stream->isReadable()) {
         return Promise\resolve('');
     }
 
+    $deferred = new Promise\Deferred();
     $buffer = '';
-    $bufferer = function ($data) use (&$buffer) {
+    $bufferer = function ($data) use (&$buffer, $deferred, $maxLength) {
         $buffer .= $data;
+        if ($maxLength !== null && isset($buffer[$maxLength])) {
+            $deferred->reject(new \OverflowException('Buffer exceeded maximum length'));
```

Comment:
LGTM, but should be added to documentation :+1:

Reply:
Will do 👍

Reply:
@clue done